honeycombio/provider.go

```diff
@@ -25,10 +25,13 @@ func Provider() *schema.Provider {
 	provider := &schema.Provider{
 		Schema: map[string]*schema.Schema{
 			"api_key": {
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("HONEYCOMBIO_APIKEY", nil),
-				Sensitive:   true,
+				Type:     schema.TypeString,
+				Required: true,
+				// Tiering goes HONEYCOMB_API_KEY > HONEYCOMBIO_APIKEY > nil
+				DefaultFunc: schema.EnvDefaultFunc(
+					"HONEYCOMB_API_KEY", schema.EnvDefaultFunc(
+						"HONEYCOMBIO_APIKEY", nil)),
```

Comment:
I'd like to see a call to `tflog.Warn()` here to notify people that they're using the deprecated env var. What do you think?

Reply:
I like that idea, let me add it.

Reply:
@jharley Ok added change, please let me know if there's anything else to update or change.